is_core/utils/__init__.py

```diff
@@ -139,87 +139,114 @@ def get_model_method_or_property_data(field_name, model, fun_kwargs):
     )
 
     if method:
-        if hasattr(method, 'field'):
-            # Generic relation
-            label = getattr(method.field, 'verbose_name', pretty_name(field_name))
-        else:
-            label = getattr(method, 'short_description', pretty_name(field_name))
+        label = getattr(method, 'short_description', pretty_name(field_name))
         try:
             return (
                 (None, label, ReadonlyWidget) if isinstance(model, type)
                 else (_get_method_or_property_value(method, field_name, model, fun_kwargs), label, ReadonlyWidget)
             )
         except InvalidMethodArguments:
             return None
-    elif hasattr(model, field_name):
+    elif class_method:
+        # Property
         return (
-            getattr(model, field_name), pretty_name(field_name), ReadonlyWidget
+            getattr(model, field_name),
+            getattr(class_method, 'short_description', pretty_name(field_name)),
+            ReadonlyWidget
         )
     else:
         return None
 
 
+def _get_model_field_label(field):
+    if field.auto_created and (field.one_to_many or field.many_to_many):
+        return (
+            getattr(field.field, 'reverse_verbose_name', None)
+            if getattr(field.field, 'reverse_verbose_name', None) is not None
+            else field.related_model._meta.verbose_name_plural
```

Comment:
`getattr(field.field, 'reverse_verbose_name', field.related_model._meta.verbose_name_plural)` is enough i believe.